### DISCUSSION TITLE
Dynamically created Enum not shown correctly in the docs

### DISCUSSION THREAD

**Discussion by major-mayer** [fastapi/fastapi#6901] [Questions] [answered]
### First check

* [x ] I added a very descriptive title to this issue.
* [ x] I used the GitHub search to find a similar issue and didn't find it.
* [x ] I searched the FastAPI documentation, with the integrated search.
* [x ] I already searched in Google "How to X in FastAPI" and didn't find any information.
* [x ] I already read and followed all the tutorial in the docs and didn't find an answer.
* [ x] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
* [ ?] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
* [x ] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).
* [x I will try] After submitting this, I commit to one of:
    * Read open issues with questions until I find 2 issues where I can help someone and add a comment to help there.
    * I already hit the "watch" button in this repository to receive notifications and I commit to help at least 2 people that ask questions in the future.
    * Implement a Pull Request for a confirmed bug.

### Example

Here's a self-contained, [minimal, reproducible, example](https://stackoverflow.com/help/minimal-reproducible-example) with my use case:

```Python
f# Python and PIP
from fastapi import FastAPI, status, Response
from pydantic import BaseModel
from enum import Enum

app = FastAPI()

class RequestModel(BaseModel):
    order_type: Enum("OrderTypeEnum", {'online': 1, 'offline': 2})


@app.post("/test")
async def test_enum(request: RequestModel):
    return request

```

### Description

* I know that in this case the Enum could be created without the functional API, but it's just to show my problem
* Open the Swagger-UI documentation and look for the RequestModel
* It is missing the different Enum values to choose from and looks like this:

![grafik](https://user-images.githubusercontent.com/17907799/110331412-61516d80-801f-11eb-8e97-3b5d9311a8d5.png)

* But the Openapi.json document looks like it could contain the Model correctly, so I'm not exactly sure if this error comes from FastAPI, Swagger UI or Pydanic, which should be responsible for mapping the model to the json schema (IIRC)

Here is what the Enum looks like in the OpenAPI schema:
![grafik](https://user-images.githubusercontent.com/17907799/110337017-bb553180-8025-11eb-9d1d-59f9229b55c6.png)
To me this seems only partially correct, because the names are missing (1 should be "online", 2 should be "offline").

### Environment

* OS: [e.g. Linux / Windows / macOS]: Linux
* FastAPI Version [e.g. 0.3.0]: 0.63.0
* Python version: 3.9

**Comment by major-mayer** [comment, marked as answer]
I found a workaround by switching from Enum to IntEnum like this:
```Python
order_type: IntEnum("OrderTypeEnum", "ONLINE OFFLINE")
```

The documentation is still a bit lacking, because it only displays a list of integers without any name or something, but it's better then nothing.
And the same thing happens if you create a IntEnum manually (non dynamic), so it could be intended. I will close this then.

**Comment by tiangolo** [comment]
Thanks for reporting back and closing the issue! ☕ 

Yep, the standards to define enums (OpenAPI, JSON Schema) currently don't have a way to define additional metadata apart from the values.

> Sorry for the long delay! 🙈  I wanted to personally address each issue/PR and they piled up through time, but now I'm checking each one in order.

**Comment by pingvincible** [comment]
Hi, everybody. Is there are plans to implement showing names for enums, especially for IntEnums, and not only array of numbers? For example, drf-spectacular can do it for DRF. It will be very cool to have such feature in FastApi.
